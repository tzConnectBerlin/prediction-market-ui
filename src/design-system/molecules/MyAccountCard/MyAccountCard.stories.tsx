import React from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import { MyAccountCard, MyAccountCardProps } from '.';

export default {
  title: 'Molecule/MyAccountCard',
  component: MyAccountCard,
} as Meta;

const Template: Story<MyAccountCardProps> = (args) => <MyAccountCard {...args} />;

export const Base = Template.bind({});
Base.args = {
  title: 'Wallet',
  walletLabel: 'Wallet',
  walletName: 'Temple',
  keyLabel: 'Key',
  address: 'tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb',
  balanceLabel: 'Balance',
  balance: 0,
};
