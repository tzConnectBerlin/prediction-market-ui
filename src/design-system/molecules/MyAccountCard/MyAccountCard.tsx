import React from 'react';
import styled from '@emotion/styled';
import { Card, Grid } from '@material-ui/core';
import { Typography } from '../../atoms/Typography';
import { lightTheme as theme } from '../../../theme';
import { Address } from '../../atoms/Address/Address';

const StyledCard = styled(Card)`
  margin: 1em;
  padding: 1em;
  display: flex;
  max-width: 30rem;
  @media (min-width: 280px) {
    width: 75%;
    flex-direction: column;
  }
  @media (min-width: 600px) {
    width: 100%;
    flex-direction: row;
  }
`;
const StyledLinkGrid = styled(Grid)`
  @media (min-width: 280px) {
    direction: row;
    justify-content: space-between;
  }
  @media (min-width: 600px) {
    margin: 1rem;
    text-align: right;
    direction: column;
    justify-content: space-between;
  }
`;

export interface MyAccountCardProps {
  /**
   * title of card
   */
  title: string;
  walletLabel: string;
  /**
   * wallet name
   */
  walletName?: string;
  keyLabel: string;
  /**
   * key/address hash
   */
  address: string;
  balanceLabel: string;
  /**
   * balance amount
   */
  balance?: string;
  /**
   *
   */
  handleAddFunds?: () => void | Promise<void>;
  /**
   *
   */
  handleDisconnect?: () => void | Promise<void>;
}
export const MyAccountCard: React.FC<MyAccountCardProps> = ({
  title,
  walletLabel,
  walletName,
  keyLabel,
  address,
  balanceLabel,
  balance,
  handleAddFunds,
  handleDisconnect,
}) => {
  return (
    <StyledCard>
      <Grid container>
        <Grid item xs={12} sm={5} marginX="1rem">
          <Typography
            color={theme.palette.text.primary}
            component="h1"
            size="1.5rem"
            marginY="1rem"
          >
            {title}
          </Typography>
          <Typography color={theme.palette.primary.main}>{walletLabel}</Typography>
          <Typography marginBottom="1rem" color={theme.palette.text.secondary}>
            {walletName}
          </Typography>
          <Typography color={theme.palette.primary.main}>{keyLabel}</Typography>
          <Address
            address={address}
            trim
            trimSize="medium"
            customStyle={{
              width: 'auto',
              marginBottom: '1rem',
              color: theme.palette.text.secondary,
            }}
          />
          <Typography color={theme.palette.primary.main}>{balanceLabel}</Typography>
          <Typography color={theme.palette.text.secondary} marginBottom="1rem">
            {balance}
          </Typography>
        </Grid>
        <Grid
          item
          container
          xs={12}
          sm={5}
          margin="1rem"
          textAlign="right"
          direction="column"
          justifyContent="space-between"
        >
          <Typography color={theme.palette.primary.main} onClick={handleDisconnect}>
            Disconnect Wallet
          </Typography>
          <Typography color={theme.palette.primary.main}>Add tez</Typography>
        </Grid>
      </Grid>
    </StyledCard>
  );
};
