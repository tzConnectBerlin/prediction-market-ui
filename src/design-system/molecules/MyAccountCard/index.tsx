export { MyAccountCard as default, MyAccountCard } from './MyAccountCard';
export type { MyAccountCardProps } from './MyAccountCard';
