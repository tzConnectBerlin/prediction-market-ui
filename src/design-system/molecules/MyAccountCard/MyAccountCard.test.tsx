import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { MyAccountCard } from './MyAccountCard';

describe('Snapshot testing MarketCard Component', () => {
  it('renders correctly with key and balance', () => {
    const Card = renderer
      .create(
        <MyAccountCard
          title="My account card"
          walletLabel="Wallet"
          walletName="Temple"
          keyLabel="Key"
          address="tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb"
          balanceLabel="Balance"
          balance={0}
        />,
      )
      .toJSON();
    expect(Card).toMatchSnapshot();
  });
});

describe('Element testing MyAccountCard Component', () => {
  it('renders correctly', async () => {
    const { getByText } = render(
      <MyAccountCard
        title="Wallet"
        walletLabel="Wallet"
        walletName="Temple"
        keyLabel="Key"
        address="tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb"
        balanceLabel="Balance"
        balance={0}
      />,
    );

    expect(getByText(/Balance/i)).toBeInTheDocument();
    expect(getByText(/Key/i)).toBeInTheDocument();
  });
});
