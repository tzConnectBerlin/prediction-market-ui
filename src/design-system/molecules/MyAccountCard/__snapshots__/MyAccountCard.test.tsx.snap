// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing MarketCard Component renders correctly with key and balance 1`] = `
.emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  margin: 1em;
  padding: 1em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 30rem;
}

@media (min-width: 280px) {
  .emotion-0 {
    width: 75%;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width: 600px) {
  .emotion-0 {
    width: 100%;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 100%;
  margin-left: 1rem;
  margin-right: 1rem;
}

@media (min-width:600px) {
  .emotion-2 {
    -webkit-flex-basis: 41.666667%;
    -ms-flex-preferred-size: 41.666667%;
    flex-basis: 41.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 41.666667%;
  }
}

.emotion-3 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  font-size: 1.5rem;
  color: rgba(0, 0, 0, 0.87);
  margin-top: 1rem;
  margin-bottom: 1rem;
  font-size: 1.5rem;
}

.emotion-3.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 4;
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: #1976d2;
}

.emotion-4.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 4;
}

.emotion-5 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  margin-bottom: 1rem;
  color: rgba(0, 0, 0, 0.6);
}

.emotion-5.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 4;
}

.emotion-7 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: auto;
  margin-bottom: 1rem;
  color: rgba(0, 0, 0, 0.6);
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-9 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.75;
  letter-spacing: 0.00938em;
}

.emotion-9.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 4;
}

.emotion-10 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  padding-top: 0.3rem;
  padding-left: 0.5rem;
}

.emotion-10.MuiSvgIcon-root {
  font-size: 1rem;
}

.emotion-12 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.6);
  margin-bottom: 1rem;
}

.emotion-12.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 4;
}

.emotion-13 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  margin: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 100%;
  margin: 1rem;
  text-align: right;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-13>.MuiGrid-item {
  max-width: none;
}

@media (min-width:600px) {
  .emotion-13 {
    -webkit-flex-basis: 41.666667%;
    -ms-flex-preferred-size: 41.666667%;
    flex-basis: 41.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 41.666667%;
  }
}

<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined emotion-1"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5 emotion-2"
    >
      <h1
        className="MuiTypography-root MuiTypography-body1  emotion-3"
        lines={4}
      >
        My account card
      </h1>
      <p
        className="MuiTypography-root MuiTypography-body1  emotion-4"
        lines={4}
      >
        Wallet
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1  emotion-5"
        lines={4}
      >
        Temple
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1  emotion-4"
        lines={4}
      >
        Key
      </p>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined emotion-7"
      >
        <div
          className="MuiGrid-root MuiGrid-item emotion-8"
        >
          <span
            className="MuiTypography-root MuiTypography-subtitle1  emotion-9"
            lines={4}
          >
            tz1VSUr...h8Cjcjb
          </span>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-10"
            copyiconsize="1rem"
            data-testid="FileCopyOutlinedIcon"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4H8c-1.1 0-1.99.9-1.99 2L6 21c0 1.1.89 2 1.99 2H19c1.1 0 2-.9 2-2V11l-6-6zM8 21V7h6v5h5v9H8z"
            />
          </svg>
        </div>
      </div>
      <p
        className="MuiTypography-root MuiTypography-body1  emotion-4"
        lines={4}
      >
        Balance
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1  emotion-12"
        lines={4}
      >
        0
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-undefined MuiGrid-direction-xs-column MuiGrid-grid-xs-12 MuiGrid-grid-sm-5 emotion-13"
    >
      <p
        className="MuiTypography-root MuiTypography-body1  emotion-4"
        lines={4}
      >
        disconnectWallet
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1  emotion-4"
        lines={4}
      >
        addTez
      </p>
    </div>
  </div>
</div>
`;
